spring-boot-starter/starter-security/src/main/java/org/camunda/bpm/spring/boot/starter/security/oauth2/CamundaSpringSecurityOAuth2AutoConfiguration.java

```diff
@@ -71,16 +80,31 @@ public FilterRegistrationBean<?> webappAuthenticationFilter() {
     filterRegistration.setDispatcherTypes(DispatcherType.REQUEST);
     return filterRegistration;
   }
-  
+
+  @Bean
+  @ConditionalOnProperty(name = "identity-provider.enabled", prefix = OAuth2Properties.PREFIX)
```

Comment:
❓ Shouldn't we specify the `havingValue`? I don't see the `OAuth2IdentityProvider` used at all for `starter-security` even if I add the property.
```suggestion
  @ConditionalOnProperty(name = "identity-provider.enabled", prefix = OAuth2Properties.PREFIX, havingValue = "true", matchIfMissing = false)
```

Reply:
It should be disabled for Spring Boot by default but enabled for Run when passing the `--oauth2` flag.

Reply:
Let's move on and fix this later.

Reply:
I created a follow-up ticket where we can improve this as well https://github.com/camunda/camunda-bpm-platform/issues/4583